homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,225 @@
+            CONF_REGION: data.get(CONF_REGION, DEFAULT_REGION),
+            CONF_TYPE: ATTR_TYPE_CLOUD,
+        }
+
+        return self.async_create_entry(title=data[CONF_USERNAME], data=auth_data)
+
+    async def _create_camera_rtsp(self, data):
+        """Create RTSP auth entry per camera in config."""
+
+        await self.async_set_unique_id(data[ATTR_SERIAL])
+        self._abort_if_unique_id_configured()
+
+        _LOGGER.debug("Create camera with: %s", data)
+
+        cam_serial = data.pop(ATTR_SERIAL)
+        data[CONF_TYPE] = ATTR_TYPE_CAMERA
+
+        return self.async_create_entry(title=cam_serial, data=data)
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return EzvizOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        # Check if ezviz cloud account is present
+        for item in self._async_current_entries():
+            if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
+                return await self.async_step_user_camera()
+
+        errors = {}
+
+        if user_input is not None:
+            if CONF_TIMEOUT not in user_input:
+                user_input[CONF_TIMEOUT] = DEFAULT_TIMEOUT
+
+            try:
+                return await self._validate_and_create_auth(user_input)
+
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+            except AbortFlow:
+                raise
+
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Optional(CONF_REGION, default=DEFAULT_REGION): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors or {}
+        )
+
+    async def async_step_user_camera(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        errors = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[ATTR_SERIAL])
+            self._abort_if_unique_id_configured()
+
+            return self.async_create_entry(
+                title=user_input[ATTR_SERIAL],
+                data={
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                    CONF_TYPE: ATTR_TYPE_CAMERA,
+                },
+            )
+
+        camera_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default=DEFAULT_CAMERA_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Required(ATTR_SERIAL): str,
```

Comment:
I'm assuming it's not possible auto detect the serial number of the camera?

Reply:
Correct. I need the serial to know what credentials match what camera.

Reply:
Actually you just gave me an idea. I could query the serials from the ezviz cloud account and trigger a multi step flow for the password on the user submitted form.

Is there any way I can trigger a reload of the main integration if someone adds a camera entry via the integration? (It needs to reload in order to rebuild the rtsp url with the correct credentials)

Reply:
I've added a discovery flow when a camera is detected without a corresponding entry. The main integration will also reload when a camera entry is added so that the camera rtsp stream make use of the credentials without additional user intervention.